src/components/Download.tsx

```diff
@@ -1,10 +1,232 @@
+          };
+        }
+        return acc;
+      }, {} as DataDictionary),
+    [columns]
+  );
+
+  const { isValid: schemaValid, errors: schemaErrors } = useMemo(() => {
+    const ajv = new Ajv();
+    const validate = ajv.compile(schema);
+    const isValid = validate(dataDictionary);
+
+    if (!isValid) {
+      const errors = validate.errors?.map((error) => error.instancePath.slice(1)) || [];
+      return { isValid: false, errors };
+    }
+
+    return { isValid: true, errors: [] };
+  }, [dataDictionary]);
+
+  const handleDownload = () => {
+    const blob = new Blob([JSON.stringify(dataDictionary, null, 2)], {
+      type: 'application/json;charset=utf-8',
+    });
+
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = `${uploadedDataTableFileName}_annotated.json`;
+    document.body.appendChild(a);
+    a.click();
+    URL.revokeObjectURL(url);
+    document.body.removeChild(a);
+  };
+
   return (
-    <div className="flex flex-col items-center">
-      <h1>Download</h1>
-      <NavigationButton label="Back - Value Annotation" viewToNavigateTo="valueAnnotation" />
+    <div className="flex flex-col items-center p-6" data-cy="download">
+      <Typography variant="h4" className="mb-6 font-bold">
+        Download
+      </Typography>
+
+      {!schemaValid && (
+        <Alert
+          severity="error"
+          data-cy="incomplete-annotations-alert"
+          className="mb-6 w-full max-w-2xl"
+        >
+          <Typography variant="h6" className="mb-2 font-bold">
+            Incomplete Annotations
+          </Typography>
+          <Typography variant="body1" className="mb-2">
+            There are incomplete annotations for columns that were mapped to variables. You can go
+            back and complete these annotations before you download the .json data dictionary.
+          </Typography>
+          <Typography variant="body1" className="mb-2">
+            The following columns have incomplete annotations:
+          </Typography>
+          <List className="list-disc pl-6" data-cy="incomplete-annotations-list">
+            {schemaErrors.map((column) => (
+              <ListItem key={column} className="list-item">
+                <Typography variant="body1">{column}</Typography>
+              </ListItem>
+            ))}
+          </List>
+          <Typography variant="body1" className="mb-2">
+            <b>NOTE</b>: If you download the .json data dictionary without completing the
+            annotations, the data dictionary will not work with subsequent neurobagel steps.
+          </Typography>
+          <Typography variant="body1" className="mb-2">
+            <b>
+              You will not be able to load the data dictionary into the annotation tool to change
+              mistakes later.
+            </b>
+          </Typography>
+        </Alert>
+      )}
+
+      {schemaValid && (
+        <Alert
```

Comment:
why not use the ternary operator here and do {schemaValid ? <alertcomponent props=validstuff> : <alertcomponent props=invalidstuff>}